packages/contracts/contracts/foundry/SphinxUtils.sol

```diff
@@ -47,6 +52,49 @@ contract SphinxUtils is SphinxConstants, StdUtils {
         return _data[_start:_end];
     }
 
+    function checkAccesses(Vm.AccountAccess[] memory accountAccesses) internal returns (bool) {
+        // Deploy the contract using create to get the deployed code
+        SphinxForkCheck forkCheck = new SphinxForkCheck();
+
+        // If there aren't two calls, (one to the deployment proxy, and another to deploy the contract)
+        // then return false.
+        if (accountAccesses.length < 2) {
+            return false;
+        }
+
+        // If the first access does not record calling deterministic deployment proxy, then return false.
+        if (accountAccesses[0].account != DETERMINISTIC_DEPLOYMENT_PROXY) {
+            return false;
+        }
+
+        // If the second access does not record deploying the contract to the correct address, then return false.
+        address expectedAddress = vm.computeCreate2Address(
+            0,
+            keccak256(type(SphinxForkCheck).creationCode),
+            DETERMINISTIC_DEPLOYMENT_PROXY
+        );
+        if (accountAccesses[1].account != expectedAddress) {
+            return false;
+        }
+
```

Comment:
Should we also check that `accountAccesses[1].accessor` is the `DETERMINISTIC_DEPLOYMENT_PROXY`? Not sure if this is important

Reply:
Yeah, we can there’s no harm. For context in a real world scenario where our fork is not installed on the users machine, the `checkFork` function is pretty much guaranteed to return false due to the first check (that the length is not correct). These additional checks are really just sanity checks that cover the possibility that someone might make a change to Foundry that causes the first check to no longer return false if our fork is not installed. In that case, we want to make sure we have sufficient coverage to confirm that Foundry is operating the way we expect it to operate. 

Technically it would be better to check as many things as possible in the `checkAccesses` field, but then we’d need tests and whatnot for all of the checks which would make this take a lot longer. I just added the set of checks that are sufficient for me to feel comfortable that if they all pass it’s extremely unlikely to be for any reason other than our fork being installed.